### PR TITLE
Markdown report script

### LINKED ISSUE #30: Ability to create markdown report
- Should be built into current reporting script
- Should be compliant with Github's flavor of Markdown

### PR DESCRIPTION
**Pull request by ajvb**
This PR adds `service_json_to_markdown_report.py`, a script for generating a Markdown report from the output of `report_json_to_service_json.py`

Along side that, this PR includes some fixes related to making the Markdown report usable / helpful for our original use case:
- Includes a new test: `test_ec2_security_group_in_use`
- Bug fix for `aws.rds.resources.rds_db_instance_tags`
- Include EBS VolumeId in test output
- Include more metadata in service json report
- Parsing of multiple comma separated values in `--aws-regions` and `--aws-profiles`
- Small error handling fix in `report_json_to_service_json.pytest_json_to_service_json`

----

I will be creating a follow-up ticket to merge `service_json_to_markdown_report.py` and `report_json_to_service_json.py` together into one report generation CLI tool.

Fixes #30 
Related to #31 

r? @g-k

### CODE REVIEW

File: service_json_to_report.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            for status in STATUSES_TO_LIST:
+                if self.test_status_counter[test+'_'+status]:
+                    print("    - [%s (%s)](#%s)" % (
+                        status,
+                        self.test_status_counter[test+'_'+status],
+                        test+'_'+status,
+                    ))
+        print("---\n")
+
+    def print_report(self):
```
Comment (ajvb) [inline service_json_to_report.py]: @g-k The nesting in this function is pretty rough. I was going to wait for a follow-up PR as this PR has already grown pretty big, but let me know if you'd like me to refactor this before we merge it in.

File: service_json_to_markdown_report.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            for status in STATUSES_TO_LIST:
+                if self.test_status_counter[test+'_'+status]:
+                    print("    - [%s (%s)](#%s)" % (
+                        status,
+                        self.test_status_counter[test+'_'+status],
+                        test+'_'+status,
+                    ))
+        print("---\n")
+
+    def print_report(self):
```
Comment (ajvb) [inline service_json_to_markdown_report.py]: @g-k The nesting in this function is pretty rough. I was going to wait for a follow-up PR as this PR has already grown pretty big, but let me know if you'd like me to refactor this before we merge it in.

(re-adding this comment, as I renamed this file and it hid the original comment hehe)

Comment (g-k) [reply service_json_to_markdown_report.py]: This is fine by me.

File: conftest.py [hunk 3]
```diff
@@ -37,19 +37,23 @@ def pytest_addoption(parser):
...
+def parse_opt(opt):
+    if not len(opt):
+        return None
+
+    if ',' in opt[0]:
+        return opt[0].split(',')
+
+    return opt
+
+
```
Comment (g-k) [inline conftest.py]: We should parse the required tags in the same way too.

File: aws/ec2/resources.py [hunk 4]
```diff
@@ -30,3 +29,25 @@ def ec2_ebs_volumes():
...
+    in_use_sec_group_ids = defaultdict(int)
+    for instance in instances:
+        for attached_sec_group in instance['SecurityGroups']:
+            in_use_sec_group_ids[attached_sec_group['GroupId']] += 1
+
+    for sec_group in sec_groups:
+        if sec_group["GroupId"] in in_use_sec_group_ids.keys():
+            sec_group["InUse"] = True
+        else:
+            sec_group["InUse"] = False
```
Comment (g-k) [inline aws/ec2/resources.py]: This is fine for now, but my personal preference is to not modify the resources to make the tests easier to understand without digging into `resources.py`.

**Pull request merged**